index.js

```diff
@@ -102,10 +102,12 @@ for (var protocol in protocols) {
           var searchname = parsedUrl.search || '';
           var hostname = parsedUrl.hostname;
           var pathname = parsedUrl.pathname;
+          var port = parsedUrl.port;
 
           var redirectOptions = typeof options === 'string' ? {} : options;
           redirectOptions.reqUrl = redirectUrl;
           redirectOptions.hostname = hostname;
+          redirectOptions.port = port;
```

Comment:
Proper testing is already paying dividends. I've discovered we were not following redirect urls correctly.